### DISCUSSION TITLE
Handling of different Tuya devices with same fingerprint

### DISCUSSION THREAD

**Discussion by jpmv27** [Koenkk/zigbee2mqtt#22313] [Devices]
I have this TuYa water leak detector: https://www.aliexpress.com/item/1005005066493360.html?spm=a2g0o.order_list.order_list_main.194.655118022Eh1mG. Until recently, it was recognized as such.
Since the change in #21247, it is now recognized as a completely different (and incorrect) device. There is even a comment in that issue from another user that also encountered this situation.
Presumably this is a manufacturing issue that we have to live with.
How common is this?
Is there a way for me to locally override the recognition mechanism to steer it towards what is correct for my case?

**Comment by Koenkk** [comment]
Could you provide the data/database.db entry of your device?

**Comment by jpmv27** [comment]
This is the entry:
```
{"id":4,"type":"EndDevice","ieeeAddr":"0xa4c138ffde6f78fe","nwkAddr":6520,"manufId":4417,"manufName":"_TZ3000_upgcbody","powerSource":"Battery","modelId":"TS0207","epList":[1],"endpoints":{"1":{"profId":260,"epId":1,"devId":1026,"inClusterList":[1,3,1280,0],"outClusterList":[25,10],"clusters":{"genBasic":{"attributes":{"stackVersion":0,"dateCode":""}},"ssIasZone":{"attributes":{"iasCieAddr":"0x5cc7c1fffedabcc4","zoneState":1}},"genPowerCfg":{"attributes":{"batteryPercentageRemaining":200}}},"binds":[{"cluster":1,"type":"endpoint","deviceIeeeAddress":"0x5cc7c1fffedabcc4","endpointID":1}],"configuredReportings":[{"cluster":1,"attrId":33,"minRepIntval":3600,"maxRepIntval":65000,"repChange":0,"manufacturerCode":null}],"meta":{}}},"appVersion":65,"stackVersion":0,"hwVersion":1,"dateCode":"","zclVersion":3,"interviewCompleted":true,"meta":{"configured":491743244},"lastSeen":1714203912444}
```

**Comment by Koenkk** [comment]
It's indeed exactly the same, however it should only be a cosmetic issue (functionality wise it uses exactly the same code)

**Comment by jpmv27** [comment]
You're correct, it's only a cosmetic issue.
Nonetheless, if I want the previous behaviour back, could I use an external converter?
Can external converters _override_ the built-in ones, or only add unknown devices?

**Reply by Koenkk** [reply]
You could stop z2m, in the database.db change `_TZ3000_upgcbody` to `_TZ3000_upgcbod`

**Comment by jpmv27** [comment]
Ah, I see. Good trick. Maybe I'll try that.
Thanks for the help.
